packages/playground/examples/canvas/main.ts

```diff
@@ -21,20 +24,78 @@ function testClick(surface: SurfaceManager, e: MouseEvent) {
   );
 }
 
+const hoverRect = document.querySelector('#hover-rect') as HTMLDivElement;
+function mouseMove(
+  surface: SurfaceManager,
+  viewportState: ViewportState,
+  e: MouseEvent
+) {
+  const [modelX, modelY] = surface.toModelCoord(e.offsetX, e.offsetY);
+  const topElement = surface.pickTop(modelX, modelY);
+
+  if (topElement) {
+    const [x, y] = surface.toViewCoord(topElement.x, topElement.y);
+
+    hoverRect.style.visibility = 'visible';
+    hoverRect.style.left = `${x}px`;
+    hoverRect.style.top = `${y}px`;
+    hoverRect.style.width = `${topElement.w * viewportState.zoom}px`;
+    hoverRect.style.height = `${topElement.h * viewportState.zoom}px`;
+  } else {
+    hoverRect.style.visibility = 'hidden';
+  }
+}
+
 function main() {
   const doc = new Y.Doc();
-  const canvas = document.querySelector('canvas') as HTMLCanvasElement;
   const yContainer = doc.getMap('container');
+
+  const canvas = document.querySelector('canvas') as HTMLCanvasElement;
+
+  const viewportState = new ViewportState();
+  const bound = canvas.getBoundingClientRect();
+  viewportState.setSize(bound.width, bound.height);
+  viewportState.setCenter(0, 0);
+
   const surface = new SurfaceManager(canvas, yContainer);
 
+  initWheelEventHandlers(surface, viewportState, canvas);
+  surface.setViewport(
```

Comment:
Opinionated speaking, I think this part feel hacky and makes the `main` too verbose.